### PR TITLE
types: Add basic types

### LINKED ISSUE #221: Add type definitions
This is more of a suggestive pull request to move to TypeScript or just set up a definition file. I would be happy to make a PR for this myself, but I want to hear your stance on typings first - if you don't want this, I'll make a PR on DefinetlyTyped instead 👍

### PR DESCRIPTION
**Pull request by hyperupcall**
Closes #221

### CODE REVIEW

File: dist/index.d.ts
Comment (JoshuaKGoldberg) [inline dist/index.d.ts:1]: [Non-actionable] I suppose it's either this or having the `.d.ts` be effectively a copy of the `.json`... but given that the file is meant to be used with `string` index keys, and I think this makes the most sense.

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM, thanks! Deferring to @muan. I have nothing to request. 🙂

File: dist/index.d.ts
Comment (hyperupcall) [reply dist/index.d.ts:1]: I think so too - I considered copying the keys verbatim, but the benefits didn't seem to outweigh the extra build step.

**Comment by muan** [comment]
Thanks!

**Pull request merged**